commit 1b0a60de665f13f5c841058f4f77632d5b3769df (refs/changes/07/3207/3)
Author: John Levon <john.levon@joyent.com>
Date:   2018-01-12T15:01:13+00:00 (1 year, 9 months ago)
    
    OS-5796 ::dtrace_options causes mdb to dump core
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/usr/src/cmd/mdb/common/modules/dtrace/dtrace.c b/usr/src/cmd/mdb/common/modules/dtrace/dtrace.c
index 10d23128aa..d2885c75a7 100644
--- a/usr/src/cmd/mdb/common/modules/dtrace/dtrace.c
+++ b/usr/src/cmd/mdb/common/modules/dtrace/dtrace.c
@@ -22,7 +22,7 @@
 /*
  * Copyright (c) 2003, 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 2013 by Delphix. All rights reserved.
- * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ * Copyright (c) 2018, Joyent, Inc. All rights reserved.
  */
 
 /*
@@ -2518,9 +2518,16 @@ static struct dtrace_options {
 	{ "aggsortkey", dtrace_options_numtostr },
 	{ "aggsortrev", dtrace_options_numtostr },
 	{ "aggsortpos", dtrace_options_numtostr },
-	{ "aggsortkeypos", dtrace_options_numtostr }
+	{ "aggsortkeypos", dtrace_options_numtostr },
+	{ "temporal", dtrace_options_numtostr },
+	{ "agghist", dtrace_options_numtostr },
+	{ "aggpack", dtrace_options_numtostr },
+	{ "aggzoom", dtrace_options_numtostr },
+	{ "zone", dtrace_options_numtostr }
 };
 
+CTASSERT(ARRAY_SIZE(_dtrace_options) == DTRACEOPT_MAX);
+
 static void
 dtrace_options_help(void)
 {
